src/sentry/profiles/task.py

```diff
@@ -365,6 +424,25 @@ def truncate_stack_needed(
         ) -> List[Any]:
             return stack
 
+    symbolicated_frames = stacktraces[0]["frames"]
+
+    if len(frames_sent) > 0:
+        raw_frames = profile["profile"]["frames"]
+        new_frames = []
+        raw_frame_idx = 0
+        frames_dict = get_frame_index_map(symbolicated_frames)
+
+        for idx in range(len(frames_sent)):
+            if idx in set(frames_sent):
```

Comment:
Can we turn `frames_sent` into a set earlier? This turns the loop `O(n^2)` instead of just `O(n)`.